src/MagiaClient.cpp

```diff
@@ -507,6 +533,36 @@ void *hook_loop(void *arguments) {
         pthread_exit(NULL);
     }
 
+    // Hook api endpoint
+    void *apiHook = lookup_symbol(libLocation, "_ZNK9UrlConfig3apiENS_3Api4TypeE"); // UrlConfig::api(UrlConfig::Api::Type)const
+    if (DobbyHook(apiHook, (void *)urlConfigApi, (void **)&urlConfigApiHooked) == RS_SUCCESS) {
+        LOGI("Successfully hooked UrlConfig::api.");
+    }
+    else {
+        initialization_error("Failed to hook UrlConfig::api.");
+        pthread_exit(NULL);
+    }
+
+    // Hook web endpoint
+    void *webHook = lookup_symbol(libLocation, "_ZNK9UrlConfig3webENS_3Web4TypeE"); // UrlConfig::web(UrlConfig::Web::Type)const
+    if (DobbyHook(webHook, (void *)urlConfigWeb, (void **)&urlConfigWebHooked) == RS_SUCCESS) {
+        LOGI("Successfully hooked UrlConfig::web.");
+    }
+    else {
+        initialization_error("Failed to hook UrlConfig::web.");
+        pthread_exit(NULL);
+    }
+
+    // Hook chat endpoint
```

Comment:
What are these for?

Reply:
That's the question I want to ask. :-) I see that when reversing libmadomagi.so. I'm curious what "chat" API is, I guess it might be some abandoned artifact, but maybe it's actually used somewhere?

Reply:
Also, I don't have a simple and reliable way to tell how many endpoint URLs are in the list. Actually I probed it with an incremental for-loop but I'm not very confident about this actually, although I didn't see non continuous `type` index number either.

Reply:
When the for-loop produces a SIGSEGV crash together with the last URL printed with the last `type` index seems meaningful, I know it probably reached the end.